modules/asset-manager-storage-fs/src/main/java/org/opencastproject/assetmanager/storage/impl/fs/OsgiFileSystemAssetStore.java

```diff
@@ -103,6 +204,61 @@ public void activate(final ComponentContext cc) throws IllegalStateException, IO
       rootDirectory = Paths.get(storageDir, DEFAULT_STORE_DIRECTORY).toFile().getAbsolutePath();
     }
     mkDirs(file(rootDirectory));
-    logger.info("Start asset manager files system store at {}", rootDirectory);
+    rootDirectories.add(rootDirectory);
+
+    // Read in multiple directories
+    int index = 1;
+    boolean isRootDirectory = true;
+    while (isRootDirectory) {
+      String directory = StringUtils.trimToNull(cc.getBundleContext().getProperty(CONFIG_STORE_ROOT_DIR + "." + index));
+
+      if (directory != null) {
+        rootDirectories.add(directory);
+      } else {
+        isRootDirectory = false;
+      }
+      index++;
+    }
+    for (String directory: rootDirectories) {
+      mkDirs(file(directory));
```

Comment:
Hoo boy, lotsa questions :D

- What if one or more defined directories
  - Cannot be created (permissions)
  - Are *secretly* a device (cf `/dev/null`)
  - Exist but aren't writable
- Should this 
  - Does it throw an error and die? (by far the simpliest)
  - Does it continue, but not use that storage target? (ideal)
- What if one or more defined directories are *subdirectories* of one (or more!) of the others

Reply:
Those ... are good questions D:

> Cannot be created (permissions)

That throws an exception

> Are secretly a device (cf /dev/null)

I don't know.

> Exist but aren't writable

I don't know, I hope that's an exception?

> Should this [...]

Throw an error and die, no? Users probably want to know if they configured unusable storage. Or if their storage has become unusable for some reason.

> What if one or more defined directories are subdirectories of one (or more!) of the others

Oh no. Why would anyone do this. Please.  
I believe this actually should work fine for the most part, but we should probably disallow it anyway? To avoid things like someone deleting their storage and accidentally also deleting all their other storages in subdirectories

Reply:
> > Are secretly a device (cf /dev/null)
> 
> I don't know. And I'm not sure how to check for or handle that.

See below? I guess this isn't really likely to happen, and at a certain point PEBKAC and there's nothing we can do... 

> > Exist but aren't writable
> 
> That won't lead to an exception. Should/Can we check this?

Maybe as part of startup we try and write something to each storage? It won't guard against storages going away during runtime, but it would ensure they start out being present at least. 

> > Should this [...]
> 
> Throw an error and die, no? Users probably want to know if they configured unusable storage. Or if their storage has become unusable for some reason.

Agree, especially for something as important as this. 

> > What if one or more defined directories are subdirectories of one (or more!) of the others
> 
> Oh no. Why would anyone do this. Please.  
> I believe this actually should work fine for the most part, but we should probably disallow it anyway? To avoid things like someone deleting their storage and accidentally also deleting all their other storages in subdirectories

Because they're ~~evil~~DevOps and made an oopsie with their ansible script. I would hope that no one would do this on purpose, but I can see it happening accidentally and then not getting noticed until things get _messy_.